README.md

```diff
@@ -1 +1,253 @@
-# vapor
\ No newline at end of file
+![Vapor Logo](docs/logo.png)
+
+# Vapors
+
+The Vapors library provides an embedded-DSL for writing logical queries over a collection of facts.
+It is based on the `cats-free` implementation of `FreeApplicative` for building the underlying algebra
+into an expression that can be optimized, evaluated, serialized, and interpreted in different ways.
+
+## Getting Started
+
+<details>
+
+### Setup
+
+1.  **Add it to your `build.sbt`**
+
+    ```
+    resolvers += Resolver.bintray("jeffmay", "maven")
+    libraryDependencies += "com.rallyhealth" %% "vapors" % "0.1.0"
+    ```
+
+2.  **Define your fact types.**
+
+    Every fact type must have a unique name as well as a Scala type. You can define a FactType with any Scala type you
+    want, but if you want to be able to use comparison operators, you must define an `Ordering`. To check equality, you
+    need to define `cats.Eq`.
+
+    ```
+    object FactTypes {
+      val DateOfBirth = FactType[LocalDate]("DateOfBirth")
+      val Role = FactType[String]("Role")
+    }
+    
+    sealed trait Role
+    object Role {
+      case object Admin extends Role
+      case object User extends Role
+      
+      implicit val eq: Eq[Role] = Eq.fromUniversalEquals
+      implicit val ordering: Ordering[Role] = Ordering.by {
+        case Admin => 1
+        case User => 2
+      }.reverse
+    }
+    ```
+
+3.  **Craft your expressions.**
+
+    You must start with either a `TerminalFactsExp` or a logical operator (such as `and` / `or`). You will typically build a
+    `TerminalFactsExp` by filtering to a specific type of fact and applying a filter condition on the instances. Since every
+    `FactType` can have multiple instances, you have to decide how to filter the list with a logical rule:
+    
+    ```
+    import cats.implicits._
+    import com.rallyhealth.vapors.factfilter.dsl._
+    
+    val isAdmin = {
+      withType(FactTypes.Role).whereAnyFactValue(_ === "admin")
```

Comment:
I understand why this is done in two steps, but I still wonder if there's another way.